pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -0,0 +1,205 @@
+	if _, err = helpers.ReconcileDaemonSet(instance, lightHouseAgent, reqLogger,
+		r.client, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.DaemonSet {
+	labels := map[string]string{
+		"app":       "submariner-lighthouse-agent",
+		"component": "lighthouse-agent",
+	}
+
+	matchLabels := map[string]string{
+		"app": appName,
+	}
+
+	allowPrivilegeEscalation := true
+	privileged := true
+	readOnlyFileSystem := false
+	runAsNonRoot := false
+	securityContext := corev1.SecurityContext{
+		Capabilities:             &corev1.Capabilities{Add: []corev1.Capability{"ALL"}},
+		AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+		Privileged:               &privileged,
+		ReadOnlyRootFilesystem:   &readOnlyFileSystem,
+		RunAsNonRoot:             &runAsNonRoot,
+	}
+
+	terminationGracePeriodSeconds := int64(0)
+
+	serviceDiscoveryDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      "submariner-lighthouse-agent",
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: matchLabels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "submariner-lighthouse-agent",
+							Image:           getImagePath(cr, serviceDiscoveryImage),
+							ImagePullPolicy: "IfNotPresent",
+							SecurityContext: &securityContext,
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "host-slash", MountPath: "/host", ReadOnly: true},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "SUBMARINER_NAMESPACE", Value: cr.Spec.Namespace},
+								{Name: "SUBMARINER_CLUSTERID", Value: cr.Spec.ClusterID},
+								{Name: "SUBMARINER_EXCLUDENS", Value: "submariner,kube-system,operators"},
+								{Name: "SUBMARINER_DEBUG", Value: strconv.FormatBool(cr.Spec.Debug)},
+								{Name: "BROKER_K8S_APISERVER", Value: cr.Spec.BrokerK8sApiServer},
+								{Name: "BROKER_K8S_APISERVERTOKEN", Value: cr.Spec.BrokerK8sApiServerToken},
+								{Name: "BROKER_K8S_REMOTENAMESPACE", Value: cr.Spec.BrokerK8sRemoteNamespace},
+								{Name: "BROKER_K8S_CA", Value: cr.Spec.BrokerK8sCA},
+							},
+						},
+					},
+					ServiceAccountName:            "submariner-operator",
+					TerminationGracePeriodSeconds: &terminationGracePeriodSeconds,
+					NodeSelector:                  map[string]string{"submariner.io/gateway": "true"},
+					HostNetwork:                   true,
+					Volumes: []corev1.Volume{
+						{Name: "host-slash", VolumeSource: corev1.VolumeSource{HostPath: &corev1.HostPathVolumeSource{Path: "/"}}},
+					},
+				},
+			},
+		},
+	}
+
+	return serviceDiscoveryDaemonSet
+}
+
+const (
+	serviceDiscoveryImage = "lighthouse-agent"
+	appName               = "submariner-lighthouse-agent"
+)
```

Comment:
Can you move these const decls to the top of the file? That's generally where folks expect to see them.

Reply:
Done.